src/main/java/com/google/sps/servlets/RecipePostServlet.java

```diff
@@ -145,33 +149,36 @@ public void editRecipe(
     String title = request.getParameter("title");
     String imgURL = getUploadedFileUrl(request, "image");
     String description = request.getParameter("description");
-    int prepTime = Integer.parseInt(request.getParameter("time"));
+    Integer prepTime = Integer.parseInt(request.getParameter("time"));
     String difficulty = request.getParameter("difficulty");
     ArrayList<String> ingredients = getIngredients(request);
     ArrayList<String> stepList = getSteps(request);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Entity recipeEntity = null;
-
     try {
       recipeEntity = getRecipeById(datastore, request.getParameter("recipeId"));
     } catch (EntityNotFoundException e) {
       request.setAttribute("error", 1);
```

Comment:
Do

request.setAttribute("error", 1);
response.sendRedirect("/");
return;

Here and remove the if in line 165, and the else in line 189 to 191.